apps/accounts/tests/test_form.py

```diff
@@ -274,3 +282,207 @@ def test_form_not_valid_if_no_explanation_or_network_data(
         form_errors = multiform.errors["__all__"]
         assert len(form_errors) > 0
         assert "no_network_no_explanation" in [error.code for error in form_errors]
+
+
+class TestCredentialForm:
+    def test_invalid_empty_credential_form(self):
+        empty_data = {
+            "link": "",
+            "file": "",
+            "type": "",
+            "public": "",
+            "title": "",
+            "description": "",
+        }
+
+        form = CredentialForm(empty_data)
+        valid_result = form.is_valid()
+        assert not valid_result
+
+    def test_valid_with_webpage_credential(self):
+        valid_page_data = {
+            "link": faker.url(),
+            "file": "",
+            "type": EvidenceType.WEB_PAGE.value,
+            "public": "on",
+            "title": " ".join(faker.words(3)),
+            "description": "",
+        }
+
+        form = CredentialForm(valid_page_data)
+        valid_result = form.is_valid()
+        assert valid_result
+
+    def test_valid_with_file_credential(self, fake_evidence):
+        valid_uploaded_doc_data = {
+            "link": "",
+            "type": EvidenceType.ANNUAL_REPORT.value,
+            "public": "on",
+            "title": " ".join(faker.words(3)),
+            "description": "",
+        }
+        file_data = {
+            "file": SimpleUploadedFile(fake_evidence.name, fake_evidence.read())
+        }
+
+        form = CredentialForm(valid_uploaded_doc_data, file_data)
+        valid_result = form.is_valid()
+        assert valid_result
+
+    def test_valid_with_file_credential(self, fake_evidence):
+        valid_uploaded_doc_data = {
+            "link": faker.url(),
+            "type": EvidenceType.ANNUAL_REPORT.value,
+            "public": "on",
+            "title": " ".join(faker.words(3)),
+            "description": "",
+        }
+        file_data = {
+            "file": SimpleUploadedFile(fake_evidence.name, fake_evidence.read())
+        }
+        form = CredentialForm(valid_uploaded_doc_data, file_data)
+        valid_result = form.is_valid()
+
+        assert not valid_result
+
+
+class TestGreenEvidenceForm:
```

Comment:
I see that you created new classes in order to group tests related to a specific form together. This confused me initially, because in other places in the codebase we use classes to designate Django-style (and unittest-style) [`TestCase`](https://docs.djangoproject.com/en/4.1/topics/testing/tools/#django.test.TestCase)-derived tests. If this additional level of organization is not strictly necessary, I suggest we stick to a pytest-style flat structure of test functions.

Reply:
I've made this change for this PR, but can we discuss this on Wednesday?

We have something like 66 classes containing tests that use pytest directly, and none of them inherit directly from any classes like `TestCase` - they're simply containers for organisation.

For what it's worth I've been loosely following this approach outlined in these conventions:

https://github.com/octoenergy/conventions/blob/master/python.md#test-folder-structure

However, I'm v happy to get an explicit convention documented that we can refer to for future PRs, becuase we have a mixture of both in the codebase, and no written guidance in our own docs right now.